### PR TITLE
Explain what happens if `flush()` fails

### PR DESCRIPTION
**Pull request by ThomasLandauer**
It should be mentioned _somewhere_ in this chapter, how to verify that the entity has really been persisted. What's the best practice here? try...catch? Or (for INSERT) check if $entity->getId() does exist? => Please expand on that.
Also, you could list all possible types of exceptions explicitly.

### CODE REVIEW

File: doctrine.rst [hunk 1]
```diff
@@ -555,6 +555,11 @@ Take a look at the previous example in more detail:
     ``Product`` objects and then subsequently call ``flush()``, Doctrine will
     execute 100 ``INSERT`` queries using a single prepared statement object.
 
+.. note::
+
+    If ``flush()`` fails, an exception is thrown, which you can catch by wrapping
```
Comment (javiereguiluz) [inline doctrine.rst]: What about replacing this:

```
If ``flush()`` fails, an exception is thrown, ...
```

by this:

```
If ``flush()`` call fails, a ``Doctrine\ORM\ORMException`` exception is thrown, ...
```

**Comment by ThomasLandauer** [comment]
Thanks, just integrated the ``Doctrine\ORM\ORMException`` hint :-)

**Comment by xabbuh** [comment]
Hm, do we really need this? Especially beginners may think that they always have to do this.

**Comment by ThomasLandauer** [comment]
Well, ***I*** was looking for this kind of explicit answer....
In my view, the docs are not a quick how-to for beginners, but also a full documentation for advanced users. Besides, the wording ("if....you *can*...") doesn't imply that everyone *should*.

**Comment by iltar** [comment]
I think you should also note that the entity manager ends up in closed state because it cannot guarantee the state of the managed entities, thus becomes utterly useless. A new instance should be made and all managed entities removed.

tl;dr while catching the exception is good as explanation, it should also document the consequences.

**Comment by weaverryan** [comment]
Thanks Thomas! I've linked to Doctrine's docs also, it's really their place to get into more details.

Cheers!
